contracts/integrations/test/framework/actors/taker.ts

```diff
@@ -61,32 +64,69 @@ export function TakerMixin<TBase extends Constructor>(Base: TBase): TBase & Cons
                 });
         }
 
-        private async *_validFillOrderCompleteFill(): AsyncIterableIterator<AssertionResult | void> {
-            const { marketMakers } = this.actor.simulationEnvironment!;
-            const assertion = validFillOrderCompleteFillAssertion(this.actor.deployment);
+        private async *_validFillOrder(): AsyncIterableIterator<AssertionResult | void> {
+            const { actors, balanceStore } = this.actor.simulationEnvironment!;
+            const assertion = validFillOrderAssertion(this.actor.deployment, this.actor.simulationEnvironment!);
             while (true) {
-                const maker = Pseudorandom.sample(marketMakers);
+                // Choose a maker to be the other side of the order
+                const maker = Pseudorandom.sample(filterActorsByRole(actors, Maker));
                 if (maker === undefined) {
-                    yield undefined;
+                    yield;
                 } else {
```

Comment:
Nit - this is a pretty long nested block. Could probably just omit the `else` and put this logic after the if smt. Or move it to a helper perhaps.